### DISCUSSION TITLE
'metadata' not passed to templates/pages

### DISCUSSION THREAD

**Discussion by slavos1** [simonw/datasette#2312] [Q&A]
Hello @simonw,
First of, thanks for Datasette.

My question: it seems the `metadata` context variable is not passed to [the custom pages](https://docs.datasette.io/en/stable/custom_templates.html#custom-pages).

When I try to render my `templates/pages/about.html`:
```jinja2
{% extends "default:index.html" %}

{% block content %}
About page.
{{super()}}
{% endblock %}
```

I get:
```
    {% block title %}{{ metadata.title or "Datasette" }}: {% for database in databases %}{{ database.name }}{% if not loop.last %}, {% endif %}{% endfor %}{% endblock %}
  File "/home/user/.local/share/pipx/venvs/datasette/lib/python3.9/site-packages/jinja2/environment.py", line 485, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'metadata' is undefined
```

I am running datasette as:
```shell
datasette \
        --static assets:static/ \
        --metadata datasette.yaml \
        --template-dir=templates/ \
        --reload -i my.db
```
and `datasette.yaml` has just my css:
```yaml
title: "My db"
extra_css_urls:
  - url: "/assets/datasette.css"
    sri: "...sha512..."
```

When I looked at [the code](https://github.com/simonw/datasette/blob/0.64.6/datasette/app.py#L1099), `metadata` was not among the template variables (`slug` was there as mentioned by [the doc](https://docs.datasette.io/en/stable/custom_templates.html#path-parameters-for-pages)).

Am I missing some setting? Thanks!

My datasette is version `0.64.6`.

**Comment by Georgekillstein** [comment]
It can be passed to the that if we can use the best out of the best [Artificial Intelligence](https://freeaiall.com/ai-news/gpt-5-will-have-a-ph-d-there-is-a-level-of-intelligence-it-is-also-fast/) in it .
